neural_compressor/common/benchmark.py

```diff
@@ -0,0 +1,528 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+import psutil
+
+from neural_compressor.common.utils import Statistics, logger
+
+description = """
+##################################################################################################################
+This is the command used to launch the Intel CPU performance benchmark, supports both Linux and Windows platform.
+To get the peak performance on Intel Xeon CPU, we should avoid crossing NUMA node in one instance.
+By default, `incbench` will trigger 1 instance on the first NUMA node.
+
+Params in `incbench`:
+ - num_instances            Default to 1.
+ - num_cores_per_instance   Default to None.
+ - C, cores                 Default to 0-${num_cores_on_NUMA-1}, decides the visible core range.
+ - cross_memory             Default to False, decides whether to allocate memory cross NUMA.
+                                Note: Use it only when memory for instance is not enough.
+
+# General use cases:
+1. `incbench main.py`: run 1 instance on NUMA:0.
+2. `incbench --num_i 2 main.py`: run 2 instances on NUMA:0.
+3. `incbench --num_c 2 main.py`: run multi-instances with 2 cores per instance on NUMA:0.
+4. `incbench -C 24-47 main.py`: run 1 instance on COREs:24-47.
+5. `incbench -C 24-47 --num_c 4 main.py`: run multi-instances with 4 COREs per instance on COREs:24-47.
+
+Note:
+    - `num_i` works the same as `num_instances`
+    - `num_c` works the same as `num_cores_per_instance`
+##################################################################################################################
+"""
+
+
+def get_linux_numa_info():
+    """Collect numa/socket information on linux system.
+
+    Returns:
+        numa_info (dict):   demo: {numa_index: {"physical_cpus": "xxx"; "logical_cpus": "xxx"}}
+                            E.g.    numa_info = {
+                                        0: {"physical_cpus": "0-23", "logical_cpus": "0-23,48-71"},
+                                        1: {"physical_cpus": "24-47", "logical_cpus": "24-47,72-95"}
+                                    }
+    """
+    result = subprocess.run(["lscpu"], capture_output=True, text=True)
```

Comment:
Consider adding a log to guide the user if `lscpu` or `numactl` fails or is not found.

Reply:
I think shell can tell user this info.